src/wasm/wasm.cpp

```diff
@@ -1123,10 +1123,39 @@ void StructSet::finalize() {
   }
 }
 
-// TODO (gc): array.new
-// TODO (gc): array.get
-// TODO (gc): array.set
-// TODO (gc): array.len
+void ArrayNew::finalize() {
+  if (rtt->type == Type::unreachable || size->type == Type::unreachable ||
+      (init && init->type == Type::unreachable)) {
+    type = Type::unreachable;
+    return;
+  }
+  type = Type(rtt->type.getHeapType(), /* nullable = */ false);
```

Comment:
Why `false` for `nullable`? Aren't we assuming all types are nullable for the moment?

Reply:
Yes, until we figure out how to do non-nullability we have to assume all types are nullable.

Reply:
Then shouldn't this be `true`?

Reply:
Oh, right, I keep mixing up non-nullability with nullability...

Reply:
Another nice cleanup we could do would be to add globally available enums for `Nullable, NonNullable` and `Mutable, NonMutable` that we could use rather than bools. I can do that if you'd like.

Reply:
Ah, good idea - yes, that would be great if you have time!